### DISCUSSION TITLE
How to remove unused import on save?

### DISCUSSION THREAD

**Discussion by scarf005** [rome/tools#4151] [Q&A]
wasn't able to find anything related on google.

**Comment by scarf005** [comment]
there's [`"source.organizeImports"`](https://stackoverflow.com/a/54660251/13503626) but it will conflict with rome.

```json
...
"editor.formatOnSave": true,
"editor.codeActionsOnSave": {
  "source.organizeImports": true
}
....
```

**Reply by leops** [reply]
If you have multiple extensions providing the `source.organizeImports` code action (like TypeScript for instance), you can set this to `source.organizeImports.rome` to explicitly specify you want the code action emitted by Rome to be applied

**Reply by scarf005** [reply]
thank you! can I set it to remove unused imports, too?

**Reply by leops** [reply]
No unfortunately at the moment this action only supports sorting import nodes, it does not detect or remove unused imports yet
